data/post-router.js

```diff
@@ -0,0 +1,105 @@
+const express = require('express');
+
+const router = express.Router();
+
+const Posts = require('./db');
+
+router.get('/', async (req, res) => {
+    try {
+        const posts = await Posts.find(req.query);
+        res.status(200).json(posts);
+    } catch (error) {
+        console.log(error);
+        res.status(500).json({
+            message: 'The posts information could not be retrieved.',
+        });
+    }
+});
+
+router.get('/:id', async (req, res) => {
+    try {
+        const post = await Posts.findById(req.params.id);
+
+        if (post) {
+            res.status(200).json(post);
+        } else {
+            res.status(404).json({ message: 'Post not found' });
+        }
+    } catch (error) {
+        console.log(error);
+        res.status(500).json({
+            message: 'The posts information could not be retrieved.',
+        });
+    }
+});
+
+router.post('/', async (req, res) => {
+    try {
+        const post = await Posts.insert(req.body);
+        res.status(201).json(post);
+    } catch (error) {
+        console.log(error);
+        res.status(500).json({
+            message: 'There was an error while saving the post to the database.',
+        });
+    }
+});
+
+router.delete('/:id', async (req, res) => {
+    try {
+        const count = await Posts.remove(req.params.id);
+        if (count > 0) {
+            res.status(200).json({ message: 'The post has been nuked.' });
+        } else {
+            res.status(404).json({ message: 'The post with the specified ID does not exist.' });
+        }
+    } catch (error) {
+        console.log(error);
+        res.status(500).json({
+            message: 'The post could not be removed.',
+        });
+    }
+});
+
+router.put('/:id', async (req, res) => {
+    try {
```

Comment:
here also you may want to check the type of data coming in to insure it matches what you expect before calling the db